assets/widget.js

```diff
@@ -20,31 +20,53 @@ function modal_actions(){
 
 	// Responsive modal
 	var $modal = $('#republication-tracker-tool-modal');
-	var $modal_content = $('#republication-tracker-tool-modal-content');
 	var $btn = $('.republication-tracker-tool-button');
 	var $close = $('.republication-tracker-tool-close');
 
+	// url hash of #show-republish: open the modal
+	if ( '#show-republish' === window.location.hash ) {
+		show_modal( $modal );
+	}
+
+	// click the republish button: open the modal
 	$btn.click(function(){
-		//$modal.html( html );
-		$modal.show();
-		$modal_content.show();
-		$('body').addClass('modal-open-disallow-scrolling');
-		$('#republication-tracker-tool-modal-content').unbind().click(function(e) {
-			e.stopPropagation();
-		});
+		show_modal( $modal );
 	});
 
+	// click on the modal: close the modal
 	$modal.click(function(){
-		$('body').removeClass('modal-open-disallow-scrolling');
-		$modal.hide();
+		close_modal( $modal );
 	});
 
+	// close button click: close the modal
 	$close.click(function(){
-		$('body').removeClass('modal-open-disallow-scrolling');
-		$modal.hide();
+		close_modal( $modal );
+	});
+
+	// escape key press: close the modal
+	$(document).keyup(function(e) {
+		if (27 === e.keyCode) {
+			close_modal( $modal );
+		}
+	});
+}
+
+function show_modal( $modal ) {
+	var $modal_content = $('#republication-tracker-tool-modal-content');
```

Comment:
`$('#republication-tracker-tool-modal-content')` is failing here since `$` is not defined inside the function or as a global. Can we just define it as a global at the top of the file and then we don't need to worry about defining it inside of each new function in here. Thanks!

Reply:
I just added the `$` instead, to keep it consistent.